src/App.tsx

```diff
@@ -1,26 +1,47 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useEffect, useState } from 'react';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+import { useAppDispatch } from './app/hooks';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { setTodos } from './features/todos';
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+export const App = () => {
+  const [isLoadingTodos, setIsLoadingTodos] = useState(false);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    setIsLoadingTodos(true);
+
+    getTodos()
+      .then((response: Todo[]) => {
+        dispatch(setTodos(response));
+      })
+      .catch(() => {
+        throw new Error("Can't get todos");
+      })
+      .finally(() => setIsLoadingTodos(false));
```

Comment:
It's better to move the logic inside the useEffect hook to a separate function, as it improves code readability, reusability, and makes testing easier. It also helps keep useEffect focused on side effects.